test/functionalTest/cases/3177_updateContext_APPEND_CPr_Registration/updateContext_APPEND_CPr_Reg.test

```diff
@@ -0,0 +1,221 @@
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# Orion Context Broker is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+#
+# For those usages not covered by this license please contact with
+# iot_support at tid dot es
+
+# VALGRIND_READY - to mark the test ready for valgrindTestSuite.sh
+
+--NAME--
+updateContext actionType APPEND with Context Provider Registration
+
+--SHELL-INIT--
+dbInit CB
+dbInit CP1
+brokerStart CB
+brokerStart CP1
+
+--SHELL--
+
+echo "01. Register CP1."
+echo "================="
+payload='{
+  "contextRegistrations": [
+    {
+      "entities": [
+        {
+          "type": "Floor",
+          "isPattern": "false",
+          "id": "Floor1"
+        }
+      ],
+      "attributes": [
+        {
+          "name": "temperature",
+          "type": "degree",
+          "isDomain": "false"
+        }
+      ],
+      "providingApplication": "http://localhost:'${CP1_PORT}'/v1"
+    }
+  ],
+  "duration": "P1M"
+}'
+orionCurl --url /v1/registry/registerContext --payload "$payload"
+echo
+echo
+
+echo "02. Creating an entity in CB with actionType APPEND by using NGSIv1."
+echo "===================================================================="
+payload='{
+    "contextElements": [
+        {
+            "type": "Floor",
+            "isPattern": "false",
+            "id": "Floor1",
+            "attributes": [
+                {
+                    "name": "temperature",
+                    "type": "degree",
+                    "value": "12"
+                }
+            ]
+        }
+    ],
+    "updateAction": "APPEND"
+}'
+orionCurl --url /v1/updateContext --payload "$payload"
+echo
+echo
+
+echo "03. Query to CP: see nothing (that's ensures the APPEND wasn't forwarded)."
+echo "=========================================================================="
+payload='{
+    "entities": [
+        {
+            "type": "Floor",
+            "isPattern": "false",
+            "id": "Floor1"
+        }
+    ],
+    "attributes": [
+        "temperature"
+    ]
+}'
+orionCurl --url /v1/queryContext --payload "$payload" --port $CP1_PORT
+echo
+echo
+
+echo "04. Query to CB: see the entity (it was created locally)."
+echo "========================================================="
+payload='{
+    "entities": [
+        {
+            "type": "Floor",
+            "isPattern": "false",
+            "id": "Floor1"
+        }
+    ],
+    "attributes": [
+        "temperature"
+    ]
+}'
+orionCurl --url /v1/queryContext --payload "$payload"
+echo
+echo
+
+

```

Comment:
Just one doubt ...
This test seems to test something that has been working for a very long time ...
Am I wrong?
Have we suddenly started to forward appends?

Reply:
I think you are right: this has worked that way time ago. However, it seems we haven't covered that case, so it's a good addition.

Actually, the PR fixes a very internal aspect of the code. I think there is no easy way to test it "directly", so it is fine if regression is still working (including the new .test).

Reply:
Fixed in b0e10f1

Reply:
Actually, I think this is most "NTC" than "Fixed in"... but it's ok anyway ;)

Reply:
Hi @fgalan Sir and @kzangeli  Sir,
Thanks for your continuous support. 
I want to know that anything is remaining from my side on this issue. If any please guide me further, so we can close this issue ASAP.
Thank you